### PR TITLE
Issue 1691

### PR DESCRIPTION
**Pull request by jarosik10**
for: https://github.com/openstax/unified/issues/1691

### CODE REVIEW

File: src/app/messages/pl/messages.json [hunk 1]
```diff
@@ -92,24 +92,24 @@
...
   "i18n:nudge:study-tools:aria-label:only-highlighting": "Ilustracja prowadzi do przycisku Moje notatki. Narzędzie, którego potrzebujesz, 100% za darmo. Ucz się sprawniej, dodając zakreślenia i notatki. Żeby zamknąć ilustrację, przejdź do albo kliknij w x.",
   "i18n:page-not-found:heading": "Ojoj, nie możemy odnaleźć tej strony.",
   "i18n:page-not-found:text-before-button": "Spróbuj wybrać inną stronę w ",
   "i18n:practice-questions:popup:close": "Kliknij, żeby zamknąć okno z Ćwiczeniami",
   "i18n:practice-questions:popup:continue": "Kontynuuj",
   "i18n:practice-questions:popup:correct": "Poprawna odpowiedź",
   "i18n:practice-questions:popup:empty:message": "Nie ma pytań do tego fragmentu książki.",
-  "i18n:practice-questions:popup:empty:next-section": "Wybierz rozdział z listy powyżej albo przejdź do",
+  "i18n:practice-questions:popup:empty:next-section": "Wybierz rozdział z listy powyżej albo przejdź do ",
```
Comment (bethshook) [inline src/app/messages/pl/messages.json:105]: Is this intentional?

Comment (jarosik10) [reply src/app/messages/pl/messages.json:105]: Yes

File: src/app/messages/pl/messages.json [hunk 2]
```diff
@@ -73,7 +73,7 @@
   "i18n:nav:profile:text": "Konto",
   "i18n:notification:appmessage:dismiss": "OK",
   "i18n:notification:appmessage:header": "Powiadomienia",
-  "i18n:notification:cookies:body": "Używamy ciasteczek, które zapewniają komfort korzystania z platformy. Pozostanie na tej stronie jest równoznaczne z wyrażeniem zgody na użycie ciasteczek. Więcej informacji oraz opcję cofnięcia udzielonej zgody znajdziesz na <a href=\"/privacy-policy\" target=\"blank\">Privacy Policy</a>",
+  "i18n:notification:cookies:body": "Używamy ciasteczek, które zapewniają komfort korzystania z platformy. Pozostanie na tej stronie jest równoznaczne z wyrażeniem zgody na użycie ciasteczek. Więcej informacji oraz opcję cofnięcia udzielonej zgody znajdziesz w <a href=\"/privacy-policy\" target=\"blank\">Polityce Prywatności</a>",
```
Comment (bethshook) [inline src/app/messages/pl/messages.json:76]: If you have time, can you also take a look at lines 34, 36, 37, 68, and 71? :grimacing:

**Comment by jarosik10** [comment]
@TomWoodward  @bethshook 
https://github.com/openstax/rex-web/blob/44a6d47c8d5ee060fe11abbdd92756f68ea4588d/src/app/messages/pl/messages.json#L36-L37
`{{style}}` always returns an English translation of highlight style - `yellow`, `green`, etc. Do we want to handle it somehow? Like making the message more complex:
```json
 "i18n:highlighter:highlight:end": "Koniec {style, select, search {zakreślenia wyniku wyszukiwania} yellow {żółtego} <...other translations> other {{style}}} zakreślenia",
```

**Comment by TomWoodward** [comment]
we already have the labels translated under keys like `"i18n:highlighting:colors:yellow": "żółty"`, i think that either:
- we re-organize the logic so that it uses the label translation separately and pokes that into the message you're referring to. not sure if formatjs has a thing for calling other keys or if we'd have to do it with code and pass it in
- we just make separate keys ` "i18n:highlighter:highlight:end:yellow"` for each label

i don't have a very strong preference either way, i am slightly nervous that composing separate translation keys will cause grammar issues in some languages

how did we solve this for spanish?

**Comment by bethshook** [comment]
Colors are not translated in Spanish either

**Comment by jarosik10** [comment]
I think that `"i18n:highlighter:highlight:end:yellow"` might be better, because to be precise it should be: _Początek  ~żółty~**żółtego** zakreślenia_. So we should not use `"żółty"` directly.

**Comment by bethshook** [comment]
Works for me. Here is the Spanish:
```
"i18n:highlighter:highlight:start:blue": "Comienzo de {select, search {resultado de búsqueda} other {texto resaltado azul}}"
"i18n:highlighter:highlight:end:blue": "Final de {select, search {resultado de búsqueda} other {texto resaltado azul}}"
"i18n:highlighter:highlight:start:green": "Comienzo de {select, search {resultado de búsqueda} other {texto resaltado verde}}"
"i18n:highlighter:highlight:end:green": "Final de {select, search {resultado de búsqueda} other {texto resaltado verde}}"
"i18n:highlighter:highlight:start:pink": "Comienzo de {select, search {resultado de búsqueda} other {texto resaltado rosado}}"
"i18n:highlighter:highlight:end:pink": "Final de {select, search {resultado de búsqueda} other {texto resaltado rosado}}"
"i18n:highlighter:highlight:start:purple": "Comienzo de {select, search {resultado de búsqueda} other {texto resaltado morado}}"
"i18n:highlighter:highlight:end:purple": "Final de {select, search {resultado de búsqueda} other {texto resaltado morado}}"
"i18n:highlighter:highlight:start:yellow": "Comienzo de {select, search {resultado de búsqueda} other {texto resaltado amarillo}}"
"i18n:highlighter:highlight:end:yellow": "Final de {select, search {resultado de búsqueda} other {texto resaltado amarillo}}"
```

**Review by bethshook — APPROVED** [review]

**Review by bethshook — APPROVED** [review]

**Pull request merged**